internal/processor/stream/processor.go

```diff
@@ -248,6 +249,14 @@ func (p *Processor) HandleStream(
 
 	for {
 		var batch model.Batch
+		var releaseBatch func()
+		if b, ok := p.batchPool.Get().(*model.Batch); ok {
+			releaseBatch = func() { p.batchPool.Put(b) }
+			batch = (*b)[:0]
+		} else {
+			releaseBatch = func() { p.batchPool.Put(&batch) }
+		}
```

Comment:
> Pool model.Batch (which is a slice), rather than a pointer to a slice. Avoids a little bit of pointer indirection.

I think the idea behind the pointer is to avoid copying the slice struct (len+cap+ptr): see https://github.com/golang/go/blob/master/src/sync/example_pool_test.go#L17-L19. I don't know if it is worth the effort, let me know what you think!

> Either call p.batchPool.Put(batch[:0]) directly, or introduce a method Processor.releaseBatch. Avoids creating heap-allocated closures.

Removed the closures 👍

Reply:
@kruskall have you benchmarked the impact of using a `model.Batch` vs `*model.Batch`? That may give us the answer we're looking for in terms of if it's worth the pointer indirection.

Reply:
I tried to go with the type value but it seems that's a linter violation.